src/common/meta/src/key.rs

```diff
@@ -80,7 +80,7 @@ use crate::DatanodeId;
 
 pub const REMOVED_PREFIX: &str = "__removed";
 
-const NAME_PATTERN: &str = "[a-zA-Z_:-][a-zA-Z0-9_:-]*";
+const NAME_PATTERN: &str = r"[a-zA-Z_:-][a-zA-Z0-9_:\-\.]*";
```

Comment:
nitpick: Looks like that `-` is also a valid character, could you add a test case for it?